### ISSUE TITLE
Switch to Python 3

### ISSUE THREAD

**Issue by joelgoop** [demokratikollen/demokratikollen#5]

**Comment by rasmuse** [comment]
OK, I have just taken the master and converted the code so it works with Python 3.4. Now Python 3.4 has to be installed in the vagrant environment and we should be ready to merge this with the master? You'll find my work so far in c1c9f8f.

**Comment by joelgoop** [comment]
It seems to be easy to use Python 3 in the Vagrant box (actually it appears to be pre-installed, but we can tell Ansible to ensure that it's there). That means that, in principle, we can just change the Ansible scripts to initialize the virtual environment using Python 3's `pyvenv-3.4` command instead of `virtualenv`. Then `python` automatically means `python-3.4` when the `pyvenv` is active. A slight problem though, is that Ubuntu 14.04 ships with a broken version of `pyvenv-3.4`. It is a known bug, as discussed [here](http://askubuntu.com/questions/488529/pyvenv-3-4-error-returned-non-zero-exit-status-1), and the solution is to initialize a `pyvenv` without `pip` and then install `pip` with a shell script. 

I haven't tried it, but the script from the thread linked above should work:

```
pyvenv-3.4 --without-pip myvenv
source ./myvenv/bin/activate
wget https://pypi.python.org/packages/source/s/setuptools/setuptools-3.4.4.tar.gz
tar -vzxf setuptools-3.4.4.tar.gz
cd setuptools-3.4.4
python setup.py install
cd ..
wget https://pypi.python.org/packages/source/p/pip/pip-1.5.6.tar.gz
tar -vzxf pip-1.5.6.tar.gz
cd pip-1.5.6
python setup.py install
cd ..
deactivate
source ./myvenv/bin/activate
```

(There was also a shorter solution, but this gets the `pip` source from `pypi` which felt more reliable.)

Maybe somebody with `bash` scripting skills can adapt it to take the virtual environment directory as an argument and check if anything else needs to be adapted. I'm also not sure what is the easiest way to include this script in the Ansible setup. @joharohl?

**Comment by joharohl** [comment]
Closed by 21d29e5e52433fb028b536c8eebe2dd924510470 and dbae982a8e250d1feb42ae8b7b681f84ba9f1d2c. I used the shorter version for the issue with `pip`. It is basically the instructions provided by pip themselves so I feel confident it should work in the future :). I took the liberty to create two virtualenvs even tough it is perhaps not necessary. So now the python3.4 is the default but there is also a python2.7 venv that has the dependencies installed as well.
